apiserver/facades/controller/caasfirewaller/mock_test.go

```diff
@@ -92,3 +87,46 @@ func (a *mockApplication) OpenedPortRanges() (network.GroupedPortRanges, error)
 	a.MethodCall(a, "OpenedPortRanges")
 	return a.appPortRanges, nil
 }
+
+type mockWatcher struct {
+	testing.Stub
+	tomb.Tomb
+}
+
+func (w *mockWatcher) Kill() {
+	w.MethodCall(w, "Kill")
+	w.Tomb.Kill(nil)
+}
+
+func (w *mockWatcher) Stop() error {
+	w.MethodCall(w, "Stop")
+	if err := w.NextErr(); err != nil {
+		return err
+	}
+	w.Tomb.Kill(nil)
+	return w.Tomb.Wait()
+}
+
+func (w *mockWatcher) Err() error {
+	w.MethodCall(w, "Err")
+	return w.Tomb.Err()
+}
+
+type mockStringsWatcher struct {
+	mockWatcher
+	changes chan []string
+}
+
+func newMockStringsWatcher() *mockStringsWatcher {
+	w := &mockStringsWatcher{changes: make(chan []string, 1)}
+	w.Tomb.Go(func() error {
+		<-w.Tomb.Dying()
+		return nil
+	})
+	return w
+}
+
+func (w *mockStringsWatcher) Changes() <-chan []string {
+	w.MethodCall(w, "Changes")
+	return w.changes
+}
```

Comment:
**nit:** We should be getting rid of these, not adding them. That will be the job in the new cycle.